### PR TITLE
feat: watsonx multimodal support

### LINKED ISSUE #2131: Image support in `WatsonxChatGenerator`
We would like to make this component support multimodal user messages.

### PR DESCRIPTION
**Pull request by ChinmayBansal**
### Related Issues

  - fixes #2131

  ### Proposed Changes:

  This PR adds multimodal (image + text) support to WatsonxChatGenerator, enabling the component to process
  both text and images in chat messages. The implementation follows established patterns from the
  AnthropicChatGenerator and LlamaCppChatGenerator multimodal support.

  **Key Features Added:**
  - Image format validation for supported formats (JPEG, PNG)
  - Proper message conversion to Watson API format with base64 data URIs
  - Support for multimodal models like meta-llama/llama-3-2-11b-vision-instruct and pixtral-12b
  - Role-based image restrictions (images only allowed in user messages)
  - Comprehensive error handling for unsupported formats and edge cases
  - Pre-validation of images before processing for better error flow

  **Implementation Details:**
  - Updated `_prepare_api_call()` method to handle multimodal content while preserving order
  - Added image format validation constants `ImageFormat` and `IMAGE_SUPPORTED_FORMATS`
  - Enhanced component docstring with detailed usage examples for multimodal scenarios
  - Added proper type annotations and removed `type: ignore` directive
  - Pre-validate all images upfront before content processing (following LlamaCpp pattern)

  ### How did you test it?

  **Unit Tests:**
  - ✅ `test_prepare_api_call_with_image()` - Tests proper multimodal message conversion
  - ✅ `test_prepare_api_call_with_unsupported_mime_type()` - Tests error handling for
  unsupported formats
  - ✅ `test_prepare_api_call_with_none_mime_type()` - Tests edge case with None mime type
  - ✅ `test_prepare_api_call_image_in_non_user_message()` - Tests role-based restrictions
  - ✅ `test_multimodal_message_processing()` - Tests end-to-end multimodal processing with mocked model
  - ✅ `test_supported_image_formats()` - Tests all supported formats (JPEG, PNG)
  - ✅ `test_multiple_images_in_single_message()` - Tests multiple image support

  **Integration Tests:**
  - ✅ `test_live_run_multimodal()` - Tests live API calls with real Watson multimodal models

  **Code Quality Verification:**
  - ✅ All linting checks pass: `hatch run fmt`
  - ✅ All type checking passes: `hatch run test:types`
  - ✅ All unit tests pass: `hatch run test:unit`

  **Manual Verification:**
  - Tested multimodal message creation and conversion
  - Verified proper error messages for validation failures
  - Confirmed Watson API format compatibility with data URI structure

  ### Notes for the reviewer

  - The implementation closely follows the patterns established in AnthropicChatGenerator and
  LlamaCppChatGenerator
  - Image validation uses the same error message format as other integrations for consistency
  - The `data:mime-type;base64,data` format with `image_url` structure is required by Watson API for multimodal processing
  - Added comprehensive test coverage that matches and exceeds the patterns used in Anthropic and LlamaCpp tests
  - All edge cases are properly handled including None mime types and role restrictions
  - Watson supports fewer image formats (JPEG, PNG only) compared to Anthropic/LlamaCpp (which also support GIF, WebP)

  ### Checklist

  - [x] I have read the [contributors
  guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the
  [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
  - [x] I have updated the related issue with new insights and changes
  - [x] I added unit tests and updated the docstrings
  - [x] I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for
  my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py [hunk 1]
```diff
@@ -263,16 +285,41 @@ def _prepare_api_call(
...
+                        supported_formats = ", ".join(IMAGE_SUPPORTED_FORMATS)
+                        msg_error = (
+                            f"Unsupported image format: {image.mime_type}. "
+                            f"WatsonX supports the following formats: {supported_formats}"
+                        )
+                        raise ValueError(msg_error)
+
+                if not msg.is_from(ChatRole.USER):
+                    msg_error = "Image content is only supported for user messages"
+                    raise ValueError(msg_error)
```
Comment (sjrl) [inline integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py]: Like in the LlamaCPP pattern could we move this higher up above current line `296` and do 
```python
    # Check that images are only in user messages
    if msg.images and not msg.is_from(ChatRole.USER):
        msg = "Image content is only supported for user messages"
        raise ValueError(msg)
```

File: integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py [hunk 2]
```diff
@@ -263,16 +285,41 @@ def _prepare_api_call(
...
+
+                content_parts: list[dict[str, Any]] = []
+                for part in msg._content:
+                    if isinstance(part, TextContent) and part.text:
+                        content_parts.append({"type": "text", "text": part.text})
+                    elif isinstance(part, ImageContent):
+                        # WatsonX expects base64 data URI format
+                        image_url = f"data:{part.mime_type};base64,{part.base64_image}"
+                        content_parts.append({"type": "image_url", "image_url": {"url": image_url}})
```
Comment (sjrl) [inline integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py]: Just for my own knowledge could you link the documentation that explains the formatting expected by WatsonX?

File: integrations/watsonx/tests/test_chat_generator.py [hunk 3]
```diff
@@ -446,3 +608,33 @@ async def test_live_run_async(self):
         assert len(results["replies"]) == 1
         assert isinstance(results["replies"][0], ChatMessage)
         assert len(results["replies"][0].text) > 0
+
+    @pytest.mark.skipif(
+        not os.environ.get("WATSONX_API_KEY") or not os.environ.get("WATSONX_PROJECT_ID"),
+        reason="WATSONX_API_KEY or WATSONX_PROJECT_ID not set",
+    )
+    def test_live_run_multimodal(self):
```
Comment (sjrl) [inline integrations/watsonx/tests/test_chat_generator.py:616]: Were you able to run this test locally?

File: integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py (hunk 2 above)
Comment (ChinmayBansal) [reply integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py]: I have linked it in the codebase. 
Here is the link for reference:  https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/fm-api-chat.html?context=wx

File: integrations/watsonx/tests/test_chat_generator.py (hunk 3 above)
Comment (ChinmayBansal) [reply integrations/watsonx/tests/test_chat_generator.py:616]: Yes I was able to run the test locally.

**Review by sjrl — APPROVED** [review]
Thanks!
